src/WinRT.Runtime/IInspectable.net5.cs

```diff
@@ -6,10 +6,12 @@ namespace WinRT
     public partial class IInspectable : IWinRTObject
     {
         IObjectReference IWinRTObject.NativeObject => _obj;
-        bool IWinRTObject.HasUnwrappableNativeObject => true;
-
-        ConcurrentDictionary<RuntimeTypeHandle, IObjectReference> IWinRTObject.QueryInterfaceCache { get; } = new();
-        ConcurrentDictionary<RuntimeTypeHandle, object> IWinRTObject.AdditionalTypeData { get; } = new();
+        bool IWinRTObject.HasUnwrappableNativeObject => true;
+
+        private Lazy<ConcurrentDictionary<RuntimeTypeHandle, IObjectReference>> _lazyQueryInterfaceCache = new();
+        ConcurrentDictionary<RuntimeTypeHandle, IObjectReference> IWinRTObject.QueryInterfaceCache => _lazyQueryInterfaceCache.Value;
```

Comment:
```suggestion
        private volatile ConcurrentDictionary<RuntimeTypeHandle, IObjectReference> = null;
        ConcurrentDictionary<RuntimeTypeHandle, IObjectReference> IWinRTObject.QueryInterfaceCache
        {
            get
            {
                var ret = _lazyQueryInterfaceCache;
                if (ret == null)
                {
                    Interlocked.CompareExchange(ref _lazyQueryInterfaceCache, new ConcurrentDictionary<RuntimeTypeHandle, IObjectReference>(), null);
                    ret = _lazyQueryInterfaceCache;
                }
                return ret;
            }
        }
```
As I have mentioned in the email conversation, `Lazy<T>` is not free and it is much cheaper to use lazy initialization using Interlocked.CompareExchange instead.

The same pattern should be used in all places that are doing the lazy initialization.

Reply:
We plan on doing that across all Lazy uses as a separate upcoming PR.

Reply:
Tracked with https://github.com/microsoft/CsWinRT/issues/1003